onnx/test/shape_inference_test.py

```diff
@@ -1920,6 +1920,30 @@ def test_tile_rank_inference(self):  # type: () -> None
             [])
         self._assert_inferred(graph, [make_tensor_value_info('y', TensorProto.FLOAT, (None, None, None))])  # type: ignore
 
+    def test_linearclassifier_1D_input(self):  # type: () -> None
+        onnx_ml = os.environ.get('ONNX_ML')  # type: ignore
```

Comment:
is this logic to determine whether to run the test or not okay with you @KeDengMS @houseroad @gramalingam ?

(We need to skip this test for ONNX_ML=0 builds as this test is for a traditional ML op)

Reply:
A better way to check it is creating a helper method using the #ifdef ONNX_ML to generate the body of the helper function, which returns whether it's build with ONNX_ML or not.

Reply:
Can you please point me to which file you think this helper can go into ? A similar example method will be useful...

Reply:
For now, I prefer to keep this logic simple and as it is. This is not too dissimilar to the logic in `gen_doc.py` where the environment variable is queried to determine which file to generate he documentation for. If this is okay for now, can I merge this PR ? @linkerzhang @houseroad